cylc/flow/config.py

```diff
@@ -271,14 +271,28 @@ def __init__(
             self.cfg['runtime']['root'] = OrderedDictWithDefaults()
 
         try:
-            parameter_values = self.cfg['scheduler']['parameters']
+            # Ugly hack to avoid templates getting included in parameters
+            parameter_values = {
+                key: value for key, value in
+                self.cfg['task parameters'].items()
+                if key != 'templates'
+            }
         except KeyError:
             # (Suite config defaults not put in yet.)
             parameter_values = {}
         try:
-            parameter_templates = self.cfg['scheduler']['parameter templates']
+            parameter_templates = self.cfg['task parameters']['templates']
+
         except KeyError:
             parameter_templates = {}
+
+        # Check that parameter templates are a section
+        if not hasattr(parameter_templates, 'update'):
+            raise SuiteConfigError(
+                '[task parameters][templates] is a section. Don\'t use it '
+                'as a parameter.'
+            )
+
```

Comment:
Won't this raise a NameError if `parameter_templates` is not defined? Also, is it not possible to use `isinstance()` instead of `hasattr()`, as the meaning would be clearer?

Reply:
If `parameter_templates` isn't defined in the config (line 284) then it is set as an empty dictionary at line 287.
I've used `hasattr` because parameter templates could be a dict or an OrderedDictWithDefaults, and this test covers both classes.

Reply:
Ah, I am blind!